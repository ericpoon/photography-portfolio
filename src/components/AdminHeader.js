import React from 'react';
import { connect } from 'react-redux';
import Header from './presentational/Header';
import { startLogout } from '../actions/auth';

export const AdminHeader = (props) => {
  return (
    <Header title={'Portfolio'}>
      <button
        onClick={() => null}
        className={'button button--no-background'}
      >
        {props.userName}
      </button>

      <button
        onClick={props.startLogout}
        className={'button button--no-background'}
      >
        Logout
      </button>
    </Header>
  );
};

const mapStateToProps = state => ({
  userName: state.auth.name,
});

export default connect(mapStateToProps, { startLogout })(AdminHeader);
