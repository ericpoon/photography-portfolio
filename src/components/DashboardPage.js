import React from 'react';

export default () => (
  <div>
    <p>Dashboard page goes here</p>
  </div>
);
