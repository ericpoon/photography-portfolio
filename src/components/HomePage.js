import React from 'react';

const HomePage = (props) => {
  const { history } = props;
  const onEnterClick = () => history.push('/portfolio');
  const onLoginClick = () => history.push('/login');

  return (
    <div className={'banner-layout'}>
      <div className={'banner-layout__banner'}>
        <h1 className={'banner-layout__subtitle'}>Firstname Lastname</h1>
        <p className={'banner-layout__title'}>Hong Kong</p>
      </div>
      <div>
        <button onClick={onEnterClick} className={'button button--larger'}>Enter</button>
      </div>
      <button onClick={onLoginClick} className={'button button--bottom-right'}>Login</button>
    </div>
  );
};

export default HomePage;
