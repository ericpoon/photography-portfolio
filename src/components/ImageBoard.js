import React from 'react';
import { DragDropContext } from 'react-dnd';
import HTML5Backend from 'react-dnd-html5-backend';
import SortableContainer from './SortableContainer';

const ROW_NUM = 3;
const COL_NUM = 3;

// todo: Item states are not persisted,
// use redux store to keep track of those images being edited (not all)

class ImageBoard extends React.Component {
  constructor(props) {
    super();
    this.state = { images: props.children };
  }

  componentWillReceiveProps(nextProps, nextContext) {
    this.setState({ images: nextProps.children });
  }

  moveCard = (dragIndex, hoverIndex) => {
    const { images } = this.state;
    const draggedCard = images[dragIndex];
    images.splice(dragIndex, 1, images[hoverIndex]);
    images.splice(hoverIndex, 1, draggedCard);
    this.setState({ images });
  };

  render() {
    const { images } = this.state;
    const { sortable } = this.props;
    const grid = [];

    images.forEach((current, index) => {
      const row = Math.floor(index / ROW_NUM);
      const col = index % COL_NUM;
      if (!grid[row]) grid[row] = [];
      grid[row][col] = sortable ? (
        <SortableContainer
          moveCard={this.moveCard}
          index={index}
          key={current.key}
        >
          {current}
        </SortableContainer>
      ) : current;
    });

    return (
      <div className={'image-card-collection'}>
        {grid.map(row => (
          <div className={'image-card-collection__row'} key={row.map(i => i.key).toString()}>
            {row.map(individual => individual)}
          </div>
        ))}
      </div>
    );
  }
}

export default DragDropContext(HTML5Backend)(ImageBoard);
