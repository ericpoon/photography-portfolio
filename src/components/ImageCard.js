import React from 'react';
import { connect } from 'react-redux';
import { addImage, editImage, deleteImage } from '../actions/images';

export class ImageCard extends React.Component {
  constructor(props) {
    super();
    const { title, subtitle, description } = props;
    this.state = { title, subtitle, description };
  }

  onTitleChange = (e) => {
    this.setState({ title: e.target.value.trim() });
  };
  onSubtitleChange = (e) => {
    this.setState({ subtitle: e.target.value.trim() });
  };
  onDescriptionChange = (e) => {
    this.setState({ description: e.target.value.trim() });
  };
  onUpdateClick = () => {
    const { title, subtitle, description } = this.state;
    const updates = { title, subtitle, description };
    this.props.editImage(this.props.id, updates);
  };
  onDiscardClick = () => {
    const { title, subtitle, description } = this.props;
    this.setState({ title, subtitle, description });
  };
  onRemoveClick = () => {
    this.props.deleteImage(this.props.id);
  };

  render() {
    const { onImageClick, editable } = this.props;
    const { title, subtitle, description } = this.state;
    if (editable) {
      return (
        <div className={'image-card'}>
          <div className={'image-card__image'} onClick={onImageClick}>
            <button
              className={'button button--floating-top-left button--danger'}
              onClick={this.onRemoveClick}
            />
          </div>
          <div>
            <input
              className={
                `image-card__title image-card__title--input${
                  this.state.title !== this.props.title ? ' is-dirty' : ''}`
              }
              value={title}
              onChange={this.onTitleChange}
            />
            <input
              className={
                `image-card__subtitle image-card__subtitle--input${
                  this.state.subtitle !== this.props.subtitle ? ' is-dirty' : ''}`
              }
              value={subtitle}
              onChange={this.onSubtitleChange}
            />
            <textarea
              className={
                `image-card__description image-card__description--textarea${
                  this.state.description !== this.props.description ? ' is-dirty' : ''}`
              }
              value={description}
              onChange={this.onDescriptionChange}
            />
            <div className={'image-card__actions'}>
              <button className={'button'} onClick={this.onUpdateClick}>Update</button>
              <button className={'button'} onClick={this.onDiscardClick}>Discard</button>
            </div>
          </div>
        </div>
      );
    }

    return (
      <div className={'image-card'}>
        <div className={'image-card__image'} onClick={onImageClick} />
        <div>
          <h3 className={'image-card__title'}>{title}</h3>
          <h4 className={'image-card__subtitle'}>{subtitle}</h4>
          <p className={'image-card__description'}>{description}</p>
        </div>
      </div>
    );
  }
}

export default connect(undefined, { addImage, editImage, deleteImage })(ImageCard);
