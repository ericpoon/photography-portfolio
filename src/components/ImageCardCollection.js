import React from 'react';

export default (props) => {
  const { children } = props;
  const grid = [];

  children.forEach((current, idx) => {
    const row = Math.floor(idx / 3);
    const col = idx % 3;
    if (!grid[row]) grid[row] = [];
    grid[row][col] = current;
  });

  return (
    <div className={'image-card-collection'}>
      {grid.map((row, idx) => (
        <div className={'image-card-collection__row'} key={row.map(i => i.key).toString()}>
          {row.map(individual => individual)}
        </div>
      ))}
    </div>
  );
};
