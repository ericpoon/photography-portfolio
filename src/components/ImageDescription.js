import React from 'react';

export default () => (
  <div>
    <h3 className={'image-card__title'}>Kowloon Tong, Hong Kong</h3>
    <h4 className={'image-card__subtitle'}>Jan 18, 2018</h4>
    <p className={'image-card__description'}>
      Putarem quodque tamquam ii ob an deumque. Fuerint judicia me assequi
      sapores ab verarum. Veniebant sex videbatur
      assignare eas una corporeis alligatus.
      Oculis nondum fusius sub ens urgeat fuerit rea sum. Eo ex deumque reddere
      publice ea similes attinet. Sensuum ac gi fallant incipit.
    </p>
  </div>
);
