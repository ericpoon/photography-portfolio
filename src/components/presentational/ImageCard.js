import React from 'react';

const ImageCard = (props) => {
  const {
    title, subtitle, description, fileUrl, onImageClick,
  } = props;

  return (
    <div className={'image-card'}>
      <div
        className={'image-card__image'}
        style={{ backgroundImage: `url(${fileUrl})` }}
        onClick={onImageClick}
      />
      <div>
        <h3 className={'image-card__title'}>{title}</h3>
        <h4 className={'image-card__subtitle'}>{subtitle}</h4>
        <p className={'image-card__description'}>{description}</p>
      </div>
    </div>
  );
};

export default ImageCard;
