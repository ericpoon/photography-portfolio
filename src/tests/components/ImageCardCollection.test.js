import React from 'react';
import { shallow } from 'enzyme';
import ImageCardCollection from '../../components/ImageCardCollection';
import { ImageCard } from '../../components/ImageCard';

let wrapper;

beforeEach(() => {
  wrapper = shallow(
    <ImageCardCollection>
      <ImageCard />
      <ImageCard />
      <ImageCard />
    </ImageCardCollection>);
});

it('should render ImageCardCollection', () => {
  expect(wrapper).toMatchSnapshot();
});
