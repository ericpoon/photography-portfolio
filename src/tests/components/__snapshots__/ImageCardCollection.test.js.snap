// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ImageCardCollection 1`] = `
<div
  className="image-card-collection"
>
  <div
    className="image-card-collection__row"
    key=",,"
  >
    <ImageCard />
    <ImageCard />
    <ImageCard />
  </div>
</div>
`;
